### ISSUE TITLE
prefer-object-rule false positives

### ISSUE THREAD

**Issue by aladdin-add** [not-an-aardvark/eslint-plugin-eslint-plugin#201]
I've just tried `eslint-plugin-eslint-plugin@4.0.0-0` in eslint repo.

```bash
/Users/weiran/repo/github/eslint/karma.conf.js
  13:18  error  Rules should be declared using the object style  eslint-plugin/prefer-object-rule

/Users/weiran/repo/github/eslint/lib/cli-engine/formatters/checkstyle.js
  31:18  error  Rules should be declared using the object style  eslint-plugin/prefer-object-rule
.....
```

We can suggest users to use `overrides` to avoid these cases?
